### PR TITLE
encrypted checkins - Apple requirement

### PR DESCRIPTION
**Pull request by hilmarf**

### CODE REVIEW

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/TraceTimeIntervalWarningService.java [hunk 1]
```diff
@@ -85,16 +91,49 @@ private int saveCheckins(List<CheckIn> checkins, Function<ByteString, byte[]> id
...
+          .saveDoNothingOnConflict(checkInProtectedReports.getLocationIdHash().toByteArray(),
+              checkInProtectedReports.getIv().toByteArray(),
+              checkInProtectedReports.getEncryptedCheckInRecord().toByteArray(),
+              submissionTimestamp);
+
+      if (checkInProtectedInsertedSuccessfully) {
+        numberOfCheckInProtectedReports++;
+      }
+    }
+    return numberOfCheckInProtectedReports;
```
Comment (hilmarf) [inline common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/TraceTimeIntervalWarningService.java:113]: should we raise an additional log message (WARN?) when the list.size() != numberOfCheckInProtectedReports

File: common/persistence/src/main/resources/db/specific/postgresql/V21__createTableForCheckInProtectedReports.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    initialization_vector BYTEA NOT NULL,
+    encrypted_check_in_record BYTEA NOT NULL,
+    submission_timestamp BIGINT NOT NULL
+);
+
+GRANT INSERT ON TABLE check_in_protected_reports TO "cwa_submission";
+GRANT USAGE, SELECT ON SEQUENCE check_in_protected_reports_id_seq TO "cwa_submission";
+
+GRANT SELECT, DELETE ON TABLE check_in_protected_reports TO "cwa_distribution";
+GRANT USAGE, SELECT ON SEQUENCE check_in_protected_reports_id_seq TO "cwa_distribution";
```
Comment (hilmarf) [inline common/persistence/src/main/resources/db/specific/postgresql/V21__createTableForCheckInProtectedReports.sql:13]: does the distribution really use the `check_in_protected_reports_id_seq` sequence?

Comment (FelixRottler) [reply common/persistence/src/main/resources/db/specific/postgresql/V21__createTableForCheckInProtectedReports.sql:13]: Not sure, I will check it

File: common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/TraceTimeIntervalWarningService.java (hunk 1 above)
Comment (FelixRottler) [reply common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/TraceTimeIntervalWarningService.java:113]: I agree, but maybe only if number of unencrypted is greater than zero? @AliceBilc can you add this?

File: common/persistence/src/main/resources/db/specific/postgresql/V21__createTableForCheckInProtectedReports.sql (hunk 2 above)
Comment (FelixRottler) [reply common/persistence/src/main/resources/db/specific/postgresql/V21__createTableForCheckInProtectedReports.sql:13]: Based on SELECT and DELETE it should not need the permissions right?

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/config/DistributionServiceConfig.java [hunk 3]
```diff
@@ -227,6 +228,14 @@ public void setAndroidQrCodePosterTemplate(QrCodePosterTemplate androidQrCodePos
...
   }
 
+  public boolean isDeprecated() {
+    return isDeprecated;
+  }
+
+  public void setDeprecated(boolean deprecated) {
+    isDeprecated = deprecated;
+  }
+
```
Comment (hilmarf) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/config/DistributionServiceConfig.java:238]: what's this for?

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/tracewarnings/structure/directory/file/TraceTimeIntervalWarningExportFileTest.java [hunk 4]
```diff
@@ -61,6 +61,7 @@ void fromTraceTimeIntervalWarningsShouldReturnNewInstance() throws Exception {
     byte[] checksum = Files.readAllBytes(outputFile.toPath().resolve(FILE_NAME_CHECKSUM));
     byte[] checksum2 = Files.readAllBytes(outputFile2.toPath().resolve(FILE_NAME_CHECKSUM));
 
+    //then
```
Comment (hilmarf) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/tracewarnings/structure/directory/file/TraceTimeIntervalWarningExportFileTest.java:64]: ?

Comment (FelixRottler) [reply services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/tracewarnings/structure/directory/file/TraceTimeIntervalWarningExportFileTest.java:64]: I see an assert statement after this comment. So I would close this comment

File: services/distribution/src/main/resources/application.yaml [hunk 5]
```diff
@@ -113,8 +113,8 @@ services:
       bucket: ${STATISTICS_FILE_S3_BUCKET:obs-cwa-public-dev}
 
     app-features:
-      - label: reserved
-        value: ${reserved:1}
+      - label: unencrypted-checkins-enabled
+        value: ${EVREG_UNENCRYPTED_CHECKINS_ENABLED:0}
```
Comment (roesslerj) [inline services/distribution/src/main/resources/application.yaml:116]: I think boolean (true/false) is easier to understand than 1/0 as is done in the submission yaml. Also decide for one.

**Review by roesslerj — DISMISSED** [review]

File: services/distribution/src/main/resources/application.yaml (hunk 5 above)
Comment (FelixRottler) [reply services/distribution/src/main/resources/application.yaml:118]: @hilmarf  why do we have that? First time I see this this config and it is not used inside distribution.

Comment (hilmarf) [reply services/distribution/src/main/resources/application.yaml:118]: it's the mobile app-feature toggle - becomes part of the mobile app-config, which is created by the distribution

Comment (FelixRottler) [reply services/distribution/src/main/resources/application.yaml:118]: ok so that means there is nothing to do here right?

Comment (hilmarf) [reply services/distribution/src/main/resources/application.yaml:118]: yes

**Review by FelixRottler — APPROVED** [review]

**Review by AliceBilc — APPROVED** [review]

**Pull request merged**